features/step_definitions/your_contributions_steps.rb

```diff
@@ -2,22 +2,35 @@
   step 'I am on step 1 of the WPCC homepage'
 end
 
-Given(/^I complete the first step and move on to second step$/) do
+Given(/^I complete the your details form and move to your contributions page$/) do
   step 'I fill in the age, gender, salary and frequency fields'
   step 'I click on "My employer makes contributions on part of my salary"'
   step 'I click the Next button'
 end
 
-When(/^I click edit on the first step section$/) do
-  find_link('edit').click
+Given(/^I complete the your details and your contributions forms and move to your results page$/) do
+  step 'I fill in the age, gender, salary and frequency fields'
+  step 'I click on "My employer makes contributions on part of my salary"'
+  step 'I click the Next button'
+  save_and_open_page
+  click_button
+end
+
+When(/^I click edit on the your details summary$/) do
+  sections = page.find_all('.section--details')
+  within(sections.first) do
+    find_link('edit').click
+  end
 end
 
-Then(/^I should return to the first step$/) do
+Then(/^I should return to the your details page$/) do
   expect(page).to have_css('#new_your_details_form')
```

Comment:
Agree with @margOnline, although this could be changed to:

```
# on the page object
class ...
    element :form, '.. ... form element/class'
end

# on the step
expect(your_details_page.form).to be_displayed # or to be_visible
```